persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/field/BsonRecordField.scala

```diff
@@ -89,25 +100,24 @@ class BsonRecordListField[OwnerType <: BsonRecord[OwnerType], SubRecordType <: B
     dbl
   }
 
-  override def setFromDBObject(dbo: DBObject): Box[List[SubRecordType]] =
-    setBox(Full(dbo.keySet.toList.map(k => {
-      valueMeta.fromDBObject(dbo.get(k.toString).asInstanceOf[DBObject])
-    })))
+  override def setFromDBObject(dbo: DBObject): Box[List[SubRecordType]] = {
+    setBox(Full(dbo.keySet.asScala.toList.map { k =>
+      valueMeta.fromDBObject(dbo.get(k).asInstanceOf[DBObject])
+    }))
+  }
 
   override def asJValue: JValue = JArray(value.map(_.asJValue))
 
   override def setFromJValue(jvalue: JValue) = jvalue match {
     case JNothing|JNull if optional_? => setBox(Empty)
-    case JArray(arr) => setBox(Full(arr.map( jv => {
+    case JArray(arr) => setBox(Full(arr.map { jv =>
       valueMeta.fromJValue(jv) openOr valueMeta.createRecord
-    })))
+    }))
     case other => setBox(FieldHelpers.expectedA("JArray", other))
   }
 
   override def setFromDocumentList(list: java.util.List[Document]): Box[List[SubRecordType]] = {
-    setBox(Full(list.map{ doc =>
-      valueMeta.fromDocument(doc)
-    }.toList))
+    setBox(Full(list.asScala.map { valueMeta.fromDocument }.toList))
```

Comment:
I'd argue that this should be broken up a little bit into multiple lines for readability. WDYT?